pkgs/development/python-modules/sourmash/default.nix

```
@@ -0,0 +1,61 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+, pythonOlder
+, rustPlatform
+, bitstring
+, cachetools
+, cffi
+, deprecation
+, matplotlib
+, numpy
+, scipy
+, screed
+, hypothesis
+, pytest-xdist
+, pyyaml
+, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "sourmash";
+  version = "4.8.2";
+  format = "pyproject";
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-4N94Ay5T7YiXdEWTO6NIHdEMfTvSbQGVEaak5tdRhHU=";
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-/BPkZ/EtB8J3s72G8ADSZHrKFi5PRNexWsboNmM+QwA=";
+  };
+
+  nativeBuildInputs = with rustPlatform; [ cargoSetupHook maturinBuildHook ];
+  propagatedBuildInputs = [ bitstring cachetools cffi deprecation matplotlib numpy scipy screed ];
```

```suggestion
  nativeBuildInputs = with rustPlatform; [
    cargoSetupHook
    maturinBuildHook
  ];

  buildInputs = [ iconv ];

  propagatedBuildInputs = [
    bitstring
    cachetools
    cffi
    deprecation
    matplotlib
    numpy
    scipy
    screed
  ];
```
we normally format things like this in python packages to increase readability. Also iconv should fix the darwin compile error. On linux it is null.